pkg/api/event.go

```diff
@@ -17,6 +18,7 @@ type Event struct {
 	Meta
 	Source         string     // MyTable
 	SourceID       string     // primary key of MyTable
+	SourceVersion  int32      // version of MyTable when the event occurred
```

Comment:
yeah, with the event record, we can ensure the eventual consistency, but we may need to consider handling this case, e.g. for two update operation, u1 and u2

- u1 update one resource firstly, and an event is trigged (e1)
- when the controller reconcile the e1, the u2 update the resource again, then a new event is trigged (e2)

we may encounter this situation: the controller get the e1 but the resource is already updated by u2, in this time the controller will send the resource that is updated by u2 (not u1), after the controller reconciled the e1,  it will continue to reconcile e2,  the controller will send the resource that is updated by u2, so the same update will be send twice

if we have a resource version in the event, when the controller get the e1 but the resource is already updated by u2, because the resource version (u1) in the event is not match with the current resource version (u2), so the controller ignore this event (e1),  then, the controller received the e2, the resource version in the event (u2) is match with the current resource version (u2), the controller sends the resource, so, with this version, we can ensure that the controller only handle the latest update

Reply:
yes, our event handling should be idempotent, currently, I think we can keep this part simple by idempotent

Reply:
remove the version from events